table/tables/tables.go

```diff
@@ -1481,18 +1481,27 @@ func CheckHandleOrUniqueKeyExistForUpdateIgnoreOrInsertOnDupIgnore(ctx context.C
 
 	// Check unique key exists.
 	{
-		for _, idx := range t.Indices() {
-			if !IsIndexWritable(idx) {
-				continue
+		canSkipIgnoreCheck := func(idx table.Index) bool {
+			if !IsIndexWritable(idx) || !idx.Meta().Unique || (t.Meta().IsCommonHandle && idx.Meta().Primary) {
+				return true
+			}
+			for _, c := range idx.Meta().Columns {
+				if modified[c.Offset] {
+					return true
```

Comment:
A bit confused, if this index related column value is changed, should we do the check?

Reply:
😵 my fault when extract method..fixed